File: app/src/main/java/com/drivingschoolrwandaapp/models/entities/Transaction.java
package com.drivingschoolrwandaapp.models.entities;

import com.google.gson.annotations.SerializedName;

import java.io.Serializable;

public class Transaction implements Serializable {
    @SerializedName("id")
    private int id;

    @SerializedName("subscriptionPlan")
    private SubscriptionPlan subscriptionPlan;

    @SerializedName("paid")
    private boolean paid;

    @SerializedName("method")
    private String method;

    @SerializedName("refId")
    private String refId;

    @SerializedName("amount")
    private double amount;

    @SerializedName("createdAt")
    private String createdAt;

    // Getters and Setters
    public int getId() { return id; }
    public SubscriptionPlan getSubscriptionPlan() { return subscriptionPlan; }
    public boolean isPaid() { return paid; }
    public String getMethod() { return method; }
    public String getRefId() { return refId; }
    public double getAmount() { return amount; }
    public String getCreatedAt() { return createdAt; }

    public String getFormattedAmount() {
        return String.format("RWF %.2f", amount);
    }
}

File: app/src/main/java/com/drivingschoolrwandaapp/models/entities/Transaction.kt
package com.drivingschoolrwandaapp.models.entities

import com.google.gson.annotations.SerializedName
import java.io.Serializable
import java.util.Locale

data class Transaction(
    @SerializedName("id") var id: Int = 0,
    @SerializedName("subscriptionPlan") var subscriptionPlan: SubscriptionPlan? = null,
    @SerializedName("paid") var paid: Boolean = false,
    @SerializedName("method") var method: String? = null,
    @SerializedName("refId") var refId: String? = null,
    @SerializedName("amount") var amount: Double = 0.0,
    @SerializedName("createdAt") var createdAt: String? = null
) : Serializable {

    fun getFormattedAmount(): String {
        return String.format(Locale.ROOT, "RWF %.2f", amount)
    }
}
